src/components/ProfilePicPopup.js

```diff
@@ -0,0 +1,83 @@
+import Popup from "./Popup";
+
+class ProfilePicPopup extends Popup {
+  constructor({ popupSelector, avatarSelector, handleFormSubmit }) {
+    super(popupSelector);
+    this._popupForm = this.popupElement.querySelector(".popup__form");
+    this._submitButton = this._popupForm.querySelector(".popup__button");
+    this._defaultButtonText = this._submitButton.textContent;
+    this._avatarElement = document.querySelector(avatarSelector);
+    this._handleFormSubmit = handleFormSubmit;
+    this._inputList = Array.from(
+      this._popupForm.querySelectorAll(".popup__input")
+    );
+    this._setEventListeners();
+  }
+
+  _getInputValues() {
+    const userInput = this._inputList.reduce((acc, input) => {
+      const { name, value } = input;
+      acc[name] = value;
+      return acc;
+    }, {});
+    return userInput;
+  }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._defaultButtonText;
+    }
+  }
+
+  _setEventListeners() {
+    this._popupForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      if (
+        this._handleFormSubmit &&
+        typeof this._handleFormSubmit === "function"
+      ) {
+        const inputValues = this._getInputValues();
+        this._handleFormSubmit(inputValues);
+        this._popupForm.reset();
+        this.close();
+      }
+    });
+    super.setEventListeners();
+  }
+
+  changeProfilePic(inputValues) {
+    this._avatarElement.src = inputValues.avatar;
```

Comment:
`UserInfo` handles avatar changing with `setNewAvatar`. You don't need to do that anywhere else